main/src/ca/uwaterloo/flix/language/phase/typer/ConstraintGen.scala

```diff
@@ -1228,4 +1278,17 @@ object ConstraintGen {
     }
   }
 
+  private def instantiateStruct(sym: Symbol.StructSym, structs: Map[Symbol.StructSym, KindedAst.Struct])(implicit flix: Flix) : (Map[Name.Label, Type], Type, Type.Var) = {
```

Comment:
I think some doc comments plus a concrete example would be helpful.

We often add some examples when something unusual is going on.

Reply:
What are your thoughts on this comment?

Reply:
Looks good.